SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebuggerEditor.cpp

```diff
@@ -0,0 +1,158 @@
+	: Super(ObjectInitializer)
+{
+	PrimaryActorTick.bCanEverTick = false;
+	PrimaryActorTick.bStartWithTickEnabled = false;
+	PrimaryActorTick.TickInterval = 1.f;
+
+	bAlwaysRelevant = false;
+	bNetLoadOnClient = false;
+	bReplicates = false;
+
+	NetUpdateFrequency = 1.f;
+
+	bShowWorkerRegions = false;
+	bIsGameInProgress = false;
+
+	OnBeginPieHandle = FEditorDelegates::BeginPIE.AddUObject(this, &ASpatialDebuggerEditor::OnPieBeginEvent);
+	OnEndPieHandle = FEditorDelegates::EndPIE.AddUObject(this, &ASpatialDebuggerEditor::OnPieEndEvent);
+}
+
+void ASpatialDebuggerEditor::BeginPlay()
+{
+	// Override parent to do nothing
+}
+
+void ASpatialDebuggerEditor::Destroyed()
+{
+	FEditorDelegates::BeginPIE.RemoveAll(this);
+	FEditorDelegates::EndPIE.RemoveAll(this);
+	DestroyWorkerRegions();
+}
+
+void ASpatialDebuggerEditor::OnPieBeginEvent(bool bIsSimulating)
+{
+	bIsGameInProgress = true;
+
+	DestroyWorkerRegions();
+
+	// Redraw editor window to show changes
+	GEditor->GetActiveViewport()->Invalidate();
+}
+
+void ASpatialDebuggerEditor::OnPieEndEvent(bool bIsSimulating)
+{
+	bIsGameInProgress = false;
+
+	if (bShowWorkerRegions && AllowWorkerBoundaries())
+	{
+		InitialiseWorkerRegions();
+		CreateWorkerRegions();
+
+		// Redraw editor window to show changes
+		GEditor->GetActiveViewport()->Invalidate();
+	}
+}
+
+void ASpatialDebuggerEditor::ShowWorkerRegions(bool bEnabled)
+{
+	bShowWorkerRegions = bEnabled;
+
+	RefreshWorkerRegions();
+}
+
+
+void ASpatialDebuggerEditor::RefreshWorkerRegions()
+{
+	DestroyWorkerRegions();
+
+	if (bShowWorkerRegions && AllowWorkerBoundaries())
+	{
+		InitialiseWorkerRegions();
+		CreateWorkerRegions();
+	}
+
+	if (GEditor != nullptr && GEditor->GetActiveViewport() != nullptr)
+	{
+		// Redraw editor window to show changes
+		GEditor->GetActiveViewport()->Invalidate();
+	}
+}
+
+bool ASpatialDebuggerEditor::AllowWorkerBoundaries() const
+{
+	// Check if multi worker is enabled and we are not in play mode.
+	UWorld* World = GetWorld();
+
+	if (World == nullptr)
+	{
+		return false;
+	}
+		
+	const ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(World->GetWorldSettings());
+
+	const bool bIsMultiWorkerEnabled = WorldSettings != nullptr && WorldSettings->IsMultiWorkerEnabled();
+
+	return bIsMultiWorkerEnabled && !bIsGameInProgress;
+	
+}
+
+void ASpatialDebuggerEditor::InitialiseWorkerRegions()
+{
+	ULayeredLBStrategy* LoadBalanceStrategy = NewObject<ULayeredLBStrategy>();
+	LoadBalanceStrategy->Init(GetWorld());
+	LoadBalanceStrategy->SetVirtualWorkerIds(1, LoadBalanceStrategy->GetMinimumRequiredWorkers());
+
+	if (LoadBalanceStrategy)
+	{
+		const ULayeredLBStrategy* LayeredLBStrategy = Cast<ULayeredLBStrategy>(LoadBalanceStrategy);
+		if (LayeredLBStrategy == nullptr)
+		{
+			UE_LOG(LogSpatialDebugger, Warning, TEXT("SpatialDebuggerEditor enabled but unable to get LayeredLBStrategy."));
+			return;
+		}
```

Comment:
This code doesn't make sense since you just created a layered LB strategy

Reply:
Good point. Removed redundant check.